cg/meta/workflow/raredisease.py

```diff
@@ -74,11 +74,39 @@ def get_sample_sheet_content_per_sample(self, case_sample: CaseSample) -> list[l
         )
         return sample_sheet_entry.reformat_sample_content
 
-    def get_workflow_parameters(self, case_id: str) -> WorkflowParameters:
+    def get_analysis_type(self, case_id: str):
+        case = self.get_case_from_string(case_id)
+        sample_analysis_type = ""
+        for link in case.links:
+            case_sample = link
+            sample_analysis_type_tmp = (
+                case_sample.sample.application_version.application.analysis_type
+            )
+            if sample_analysis_type_tmp != sample_analysis_type and sample_analysis_type != "":
+                raise ValueError(
+                    f"{sample_analysis_type_tmp} has not the same analysis type as other samples in the case"
+                )
+            sample_analysis_type = sample_analysis_type_tmp
+        return sample_analysis_type
```

Comment:
We already have `get_data_analysis_type` in the analysis API.

You could also add the multiple analysis type check there and raise a `ValueError`

```suggestion
```

Reply:
Agreed, and homogenize the function with MIP. I just want to check with @Vince-janv whether fetching from lims or statusDB is the most clever thing to do in light of the future hospital cluster integration (i.e. if we should limit the access to lims as much as possible)

Reply:
This method is also used for the delivery report, so we should check if there's a chance the value from LIMS differs from the one in the application view of StatusDB. It seems unlikely, but we've done odd things before with other fields 🤷